File: src/models/game.rs
use crate::models::universe::galaxy::Galaxy;
use crate::models::ships::ship::Ship;

pub struct Game {
    pub player_ship: Ship,
    pub galaxy: Galaxy,
    pub ally_ships: Vec<Ship>,
    pub neutral_ships: Vec<Ship>,
    pub hostile_ships: Vec<Ship>,
    pub game_progress:usize
}

File: entities/game.go
package entities

import (
	"github.com/Steelstone3/Star-Trek-Explorers/entities/ships"
	"github.com/Steelstone3/Star-Trek-Explorers/entities/worlds"
)

type Game struct {
	PlayerShip      ships.Ship
	FederationShips []ships.Ship
	KlingonShips    []ships.Ship
	Universe        worlds.Galaxy
}

func ConstructGame() Game {
	return Game{
		PlayerShip:      ships.ConstructFederationShip(1),
		FederationShips: ships.ConstructRandomFederationShips(),
		KlingonShips:    ships.ConstructRandomAiKlingonShips(),
		Universe:        worlds.ConstructGalaxy(),
	}
}
